staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -426,6 +426,36 @@ func (c *Cacher) startCaching(stopChannel <-chan struct{}) {
 	}
 }
 
+// isReady returns true if the cacher is ready, that means that the channel is closed
+func (c *Cacher) isReady() bool {
+	select {
+	case <-c.readyCh:
+		return true
+	default:
+		return false
+	}
+}
+
+// set cacher to ready
+func (c *Cacher) unsetReady() {
+	select {
+	case <-c.readyCh:
+		c.readyCh = make(chan struct{})
```

Comment:
hmm - this isn't thread-safe, right?

Reply:
I'm personally not against synchronizing using channels, but I think I would rather:

- leave the concept `ready` type as it was
- leave the locking inside it (as we need it to make things like unsetReady thread-safe)
- switch the internal mechanism of it to use channel

Reply:
to be honest, I thought that this was thread-safe but now I'm not sure
```
	select {
	case <-c.readyCh:
		c.readyCh = make(chan struct{})
```

I found the concept of ready easy to follow too, but the dependency on stop make the sync.Cond complex ... let me try again